tests/TestHarness/Cluster.py

```diff
@@ -999,7 +999,7 @@ def activateInstantFinality(self, launcher, biosFinalizer, pnodes):
         # call setfinalizer
         numFins = 0
         for n in launcher.network.nodes.values():
-            if n.keys[0].blspubkey is None:
+            if len(n.keys) == 0 or n.keys[0].blspubkey is None:
```

Comment:
no big deal, but I think this is more idiomatic:
```suggestion
            if not n.keys or not n.keys[0].blspubkey:
```